packages/backend/src/app.ts

```diff
@@ -267,26 +237,40 @@ export class App {
       }
     })
 
-    this.koa.use(this.apolloServer.getMiddleware())
-  }
+    koa.use(this.apolloServer.getMiddleware())
 
-  private _isAdmin(ctx: Koa.Context): boolean {
-    return ctx.request.header['x-api-key'] == this.config.adminKey
+    this.adminServer = koa.listen(port)
   }
 
-  private async _getSession(ctx: Koa.Context): Promise<Session | undefined> {
-    const key = ctx.request.header.authorization || ''
-    if (key && key.length) {
-      const sessionService = await this.container.use('sessionService')
-      return await sessionService.get(key)
-    } else {
-      return undefined
-    }
-  }
+  public async startOpenPaymentsServer(port: number | string): Promise<void> {
+    const koa = new Koa<DefaultState, AppContext>()
 
-  private async _setupRoutes(): Promise<void> {
-    this.publicRouter.use(bodyParser())
-    this.publicRouter.get('/healthz', (ctx: AppContext): void => {
+    koa.context.container = this.container
+    koa.context.closeEmitter = await this.container.use('closeEmitter')
+    koa.context.logger = await this.container.use('logger')
+
+    koa.use(
+      async (
+        ctx: {
+          status: number
+          set: (arg0: string, arg1: string) => void
+          body: string
+        },
+        next: () => void | PromiseLike<void>
+      ): Promise<void> => {
+        if (this.isShuttingDown) {
+          ctx.status = 503
+          ctx.set('Connection', 'close')
+          ctx.body = 'Server is in the process of restarting'
+        } else {
+          return next()
+        }
+      }
+    )
```

Comment:
This could be moved to a separate function that's called by both `startAdminServer` and `startOpenPaymentsServer`